conans/client/build/cppstd_flags.py

```diff
@@ -35,7 +35,20 @@ def cppstd_flag(compiler, compiler_version, cppstd):
     return flag
 
 
-def cppstd_default(compiler, compiler_version):
+def cppstd_flag_new(settings):
+    compiler = settings.get_safe("compiler")
+    compiler_version = settings.get_safe("compiler.version")
+    cppstd = cppstd_from_settings(settings)
+    return cppstd_flag(compiler, compiler_version, cppstd)
+
+
+def cppstd_default(settings):
```

Comment:
I would have prefer not to touch the ``cppstd_default``, i don't see how it is related. The feature is to expose ``cppstd_flag`` tool.

Reply:
The `cppstd_default` part could be covered by https://github.com/conan-io/conan/pull/6713

Reply:
the reason is the same and explained in PR - default cppstd may depend on **compiler.base** (and other settings, in theory), so the current interface is unsuitable for the Intel compiler.
see https://software.intel.com/en-us/cpp-compiler-developer-guide-and-reference-std-qstd
```
-std=gnu++98 (default for C++) (GCC base)
/Qstd | OFF (Visual Studio base)
```